webots_ros2_driver/webots_ros2_driver/webots_launcher.py

```diff
@@ -82,3 +124,57 @@ def __init__(self, output='screen', world=None, gui=True, mode='realtime', strea
             ],
             **kwargs
         )
+
+    def execute(self, context: LaunchContext):
+        # Check if the user wants to convert URDF files into robots
+        if self.__robots:
+            worldPath = self.__world.perform(context)
+            if not worldPath:
+                sys.exit('World file not specified (has to be specified with world=path/to/my/world.wbt')
+
+            worldCopy = worldPath[:-4] + '_with_URDF_robot.wbt'
+            shutil.copyfile(worldPath, worldCopy)
```

Comment:
I am afraid this will not work as a package might be installed in the`/opt` directory for which we need `sudo` privileges. Instead, we should put it in the `/temp` directory (there is a Python API for getting a path to a temporary directory).

Reply:
Is it possible to spawn a robot into a world? Instead of 'hardcoding' it in the worldfiles?

Reply:
We can make it possible. Why do you think it is an advantage?

Reply:
In my opinion robots operate in a world, instead of being part of it. But that might just be semantics.

But a typical flow for us is to have a repo with worlds (our office). And several of our robots can operate in this world.
Now we need N worlds for the N robot types we have. (Or script our way around that).

Reply:
No, you don't need N worlds for the N robot types, one world will be enough. It will generate new worlds in the background, no need to worry about that. The only difference is that you don't need a spawner.

From user's perspective it will be:
```py
webots = WebotsLauncher(
    world=PathJoinSubstitution([package_dir, 'worlds', world]),
    robots=[{'urdf': '/path/to/urdf.urdf', 'translation': '0 0 0'}]
)
```

instead of:
```py
webots = WebotsLauncher(
    world=PathJoinSubstitution([package_dir, 'worlds', world])
)
```

Reply:
Ah true, I had the previous situation in mind. So this effort already helps!

Than the typical use-case is covered.

Only thing left on my double agenda: we have a 'game' going on between colleagues where we can race each other in webots. It would be nice if anyone could 'join' a world without having to restart the world for all players.

Reply:
> Only thing left on my double agenda: we have a 'game' going on between colleagues where we can race each other in webots. It would be nice if anyone could 'join' a world without having to restart the world for all players.

I would love to see the game :)

You can still create a supervisor controller which will spawn robots on demand. It is more effort though. We assumed it is a very uncommon use-case, so figured out it is not very important to support it out-of-the-box. Can you think of another use-case in which it is a common practice to spawn a robot dynamically?

Reply:
Is quickly iterating on your robot a use-case?

You can leave the (rather heavy) world running in webots and continue developing on the robot. 
Continuously restarting webots takes time unfortunately.